File: 3351-MaximizeHappinessOfSelectedChildren/3351-MaximizeHappinessOfSelectedChildren.cpp
// Last updated: 6/11/2026, 11:18:40 AM
class Solution {
public:
    long long maximumHappinessSum(vector<int>& arr, int k) {
        long long tk = 0;
        priority_queue<long long> pq;
        for(long long el: arr) pq.push(el);

        long long tsum = 0;
        while(!pq.empty()){
            long long tel = pq.top(); pq.pop();
            tsum += max(0LL, (tel-tk));
            tk++;
            if(tk == k) break;
        }
        return tsum;
    }
};

File: 3351-MaximizeHappinessOfSelectedChildren/3351-MaximizeHappinessOfSelectedChildren.py
# Last updated: 6/11/2026, 11:18:46 AM
class Solution:
    def maximumHappinessSum(self, happiness: List[int], k: int) -> int:
        happiness.sort() 
        sum1 = 0
        delay = 0
        for i in range(k):
            curr = happiness[-(i+1)]
            # print(curr)
            sum1 += max(0,curr - delay)
            # print(sum1)
            delay+=1
        return sum1
            
        